timesketch/lib/datastores/opensearch.py

```diff
@@ -1271,3 +1296,505 @@ def flush_queued_events(self, retry_count=0):
+
+                try:
+                    response = (
+                        self.client.search(  # pylint: disable=unexpected-keyword-arg
+                            body=query, request_timeout=worker_request_timeout
+                        )
+                    )
+                except RequestError as re:
+                    os_logger.error(
+                        "[Slice %s/%s] RequestError: %s. "
+                        "PIT might have expired or query issue.",
+                        log_slice_id,
+                        num_slices,
+                        str(re),
+                    )
+                    break
+                except ConnectionTimeout:
+                    os_logger.warning(
+                        "[Slice %s/%s] Connection timeout during search. "
+                        "Stopping slice.",
+                        log_slice_id,
+                        num_slices,
+                    )
+                    break
+                except Exception as e:  # pylint: disable=broad-exception-caught
+                    os_logger.error(
+                        "[Slice %s/%s] Error during search: %s",
+                        log_slice_id,
+                        num_slices,
+                        str(e),
+                        exc_info=True,
+                    )
+                    # Signal other threads and main to stop due to unexpected error
+                    stop_event.set()
+                    break
+
+                hits = response.get("hits", {}).get("hits", [])
+                if not hits:
+                    os_logger.info(
+                        "[Slice %s/%s] No more documents.", log_slice_id, num_slices
+                    )
+                    break
+
+                for hit in hits:
+                    if stop_event.is_set():
+                        break
+                    if "_source" in hit:
+                        event_data_to_export = {
+                            **hit["_source"],
+                            "_id": hit.get("_id"),
+                            "_index": hit.get("_index"),
+                        }
+                        # Try to put item onto the queue with timeout
+                        # This loop handles backpressure if the queue is full.
+                        while not stop_event.is_set():
+                            try:
+                                output_queue.put(
+                                    event_data_to_export, timeout=1
+                                )  # Short timeout for responsiveness
+                                break  # Item put successfully
+                            except queue.Full:
+                                # Queue is full, wait a bit and retry.
+                                # stop_event is checked at the start of this inner loop.
+                                time.sleep(0.1)  # Avoid busy-waiting
+                            except (  # pylint: disable=broad-exception-caught
```

Comment:
is there a way to refactor this so the code does not have to be that much indented and readability is improved?

Reply:
Let me try putting some logic into a helper function. Maybe that helps to make the code more readable.